### ISSUE TITLE
Leo devel from GitHub is not working on Debian

### ISSUE THREAD

**Issue by ranvik14** [leo-editor/leo-editor#1848]
For the background of this issue, see these conversation entries on Leo's Google Group:

- https://groups.google.com/g/leo-editor/c/Uaug4xW21RI/m/48yzQ8JlAAAJ

- https://groups.google.com/g/leo-editor/c/Uaug4xW21RI/m/XScZf2jJAAAJ

###

I followed the description provided in the documentation related to **"Installing Leo with git"**.

- http://leoeditor.com/installing.html#installing-leo-with-git

For Step 3 : **"Install dependencies and create leo script launcher"** I received the following error:

`(Leo) user@debian-leo-vm:~/PyVE/Repo/Leo$ cd leo-editor`
`(Leo) user@debian-leo-vm:~/PyVE/Repo/Leo/leo-editor$ pip3 install --editable /home/user/PyVE/Repo/Leo/leo-editor`
Obtaining file:///home/user/PyVE/Repo/Leo/leo-editor
  Installing build dependencies ... error
  Complete output from command /home/user/PyVE/Repo/Leo/bin/python3 -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-_mtbzqd1 --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools>=38.6.0 setupext-janitor>=1.1 wheel>=0.31.0 twine>=1.11.0:
  Collecting setuptools>=38.6.0
    Using cached https://files.pythonhosted.org/packages/60/6a/dd9533ae9367a1f35b1b704e2ac9554a61e6dfa4d97e1d86c3760e1e135d/setuptools-54.1.1-py3-none-any.whl
  Collecting setupext-janitor>=1.1
    Using cached https://files.pythonhosted.org/packages/fe/7c/21ab50164321e657d7ca281bc827b317351b0504dca72025b1e8063d4b91/setupext_janitor-1.1.2-py2.py3-none-any.whl
  Collecting wheel>=0.31.0
    Using cached https://files.pythonhosted.org/packages/65/63/39d04c74222770ed1589c0eaba06c05891801219272420b40311cd60c880/wheel-0.36.2-py2.py3-none-any.whl
  Collecting twine>=1.11.0
    Using cached https://files.pythonhosted.org/packages/eb/d7/a46b7eb07b491a9295c3581f86ad68f97df83e000dc9a4c6e3c710c432dd/twine-3.3.0-py3-none-any.whl
  Collecting requests-toolbelt!=0.9.0,>=0.8.0 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/60/ef/7681134338fc097acef8d9b2f8abe0458e4d87559c689a8c306d0957ece5/requests_toolbelt-0.9.1-py2.py3-none-any.whl
  Collecting pkginfo>=1.4.2 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/b0/9a/d78e7c299eb5659bc3a036e5a968a399c62bfe0b2aa18baf7d13f43373ba/pkginfo-1.7.0-py2.py3-none-any.whl
  Collecting rfc3986>=1.4.0 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/78/be/7b8b99fd74ff5684225f50dd0e865393d2265656ef3b4ba9eaaaffe622b8/rfc3986-1.4.0-py2.py3-none-any.whl
  Collecting importlib-metadata; python_version < "3.8" (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/34/7c/15499080be7cbb30579630238c6472c7c3529deca03bfab7bdead6419cd1/importlib_metadata-3.7.2-py3-none-any.whl
  Collecting keyring>=15.1 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/4a/4f/a63f284d26eac29a29e24657d12f17bddac58b12673ca39b53ae5f20c1a3/keyring-23.0.0-py3-none-any.whl
  Collecting requests>=2.20 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/29/c1/24814557f1d22c56d50280771a17307e6bf87b70727d975fd6b2ce6b014a/requests-2.25.1-py2.py3-none-any.whl
  Collecting tqdm>=4.14 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/f8/3e/2730d0effc282960dbff3cf91599ad0d8f3faedc8e75720fdf224b31ab24/tqdm-4.59.0-py2.py3-none-any.whl
  Collecting readme-renderer>=21.0 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/39/a5/459adfa22ea237f6e8d0fa95ad29d7369579a5eec26f016ab34bb7f8359c/readme_renderer-29.0-py2.py3-none-any.whl
  Collecting colorama>=0.4.3 (from twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/44/98/5b86278fbbf250d239ae0ecb724f8572af1c91f4a11edf4d36a206189440/colorama-0.4.4-py2.py3-none-any.whl
  Collecting typing-extensions>=3.6.4; python_version < "3.8" (from importlib-metadata; python_version < "3.8"->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/60/7a/e881b5abb54db0e6e671ab088d079c57ce54e8a01a3ca443f561ccadb37e/typing_extensions-3.7.4.3-py3-none-any.whl
  Collecting zipp>=0.5 (from importlib-metadata; python_version < "3.8"->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/0f/8c/715c54e9e34c0c4820f616a913a7de3337d0cd79074dd1bed4dd840f16ae/zipp-3.4.1-py3-none-any.whl
  Collecting jeepney>=0.4.2; sys_platform == "linux" (from keyring>=15.1->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/51/b0/a6ea72741aaac3f37fb96d195e4ee576a103c4c04e279bc6b446a70960e1/jeepney-0.6.0-py3-none-any.whl
  Collecting SecretStorage>=3.2; sys_platform == "linux" (from keyring>=15.1->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/d9/1e/29cd69fdac7391aa51510dfd42aa70b4e6a826c8cd019ee2a8ab9ec0777f/SecretStorage-3.3.1-py3-none-any.whl
  Collecting certifi>=2017.4.17 (from requests>=2.20->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/5e/a0/5f06e1e1d463903cf0c0eebeb751791119ed7a4b3737fdc9a77f1cdfb51f/certifi-2020.12.5-py2.py3-none-any.whl
  Collecting chardet<5,>=3.0.2 (from requests>=2.20->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl
  Collecting urllib3<1.27,>=1.21.1 (from requests>=2.20->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/23/fc/8a49991f7905261f9ca9df5aa9b58363c3c821ce3e7f671895442b7100f2/urllib3-1.26.3-py2.py3-none-any.whl
  Collecting idna<3,>=2.5 (from requests>=2.20->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/a2/38/928ddce2273eaa564f6f50de919327bf3a00f091b5baba8dfa9460f3a8a8/idna-2.10-py2.py3-none-any.whl
  Collecting docutils>=0.13.1 (from readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/81/44/8a15e45ffa96e6cf82956dd8d7af9e666357e16b0d93b253903475ee947f/docutils-0.16-py2.py3-none-any.whl
  Collecting bleach>=2.1.0 (from readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/f0/46/2bbd92086a4c6f051214cb48df6d9132b5f32c5e881d3f4991b16ec7e499/bleach-3.3.0-py2.py3-none-any.whl
  Collecting six (from readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/ee/ff/48bde5c0f013094d729fe4b0316ba2a24774b3ff1c52d924a8a4cb04078a/six-1.15.0-py2.py3-none-any.whl
  Collecting Pygments>=2.5.1 (from readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/3a/80/a52c0a7c5939737c6dca75a831e89658ecb6f590fb7752ac777d221937b9/Pygments-2.8.1-py3-none-any.whl
  Collecting cryptography>=2.0 (from SecretStorage>=3.2; sys_platform == "linux"->keyring>=15.1->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/fa/2d/2154d8cb773064570f48ec0b60258a4522490fcb115a6c7c9423482ca993/cryptography-3.4.6.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
  Collecting packaging (from bleach>=2.1.0->readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/3e/89/7ea760b4daa42653ece2380531c90f64788d979110a2ab51049d92f408af/packaging-20.9-py2.py3-none-any.whl
  Collecting webencodings (from bleach>=2.1.0->readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/f4/24/2a3e3df732393fed8b3ebf2ec078f05546de641fe1b667ee316ec1dcf3b7/webencodings-0.5.1-py2.py3-none-any.whl
  Collecting cffi>=1.12 (from cryptography>=2.0->SecretStorage>=3.2; sys_platform == "linux"->keyring>=15.1->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/97/2d/cd29c79f2eb1384577d0662f23c89d29621152f14bef8c6b25747785744b/cffi-1.14.5-cp37-cp37m-manylinux1_x86_64.whl
  Collecting pyparsing>=2.0.2 (from packaging->bleach>=2.1.0->readme-renderer>=21.0->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/8a/bb/488841f56197b13700afd5658fc279a2025a39e22449b7cf29864669b15d/pyparsing-2.4.7-py2.py3-none-any.whl
  Collecting pycparser (from cffi>=1.12->cryptography>=2.0->SecretStorage>=3.2; sys_platform == "linux"->keyring>=15.1->twine>=1.11.0)
    Using cached https://files.pythonhosted.org/packages/ae/e7/d9c3a176ca4b02024debf82342dab36efadfc5776f9c8db077e8f6e71821/pycparser-2.20-py2.py3-none-any.whl
  Building wheels for collected packages: cryptography
    Running setup.py bdist_wheel for cryptography: started
    Running setup.py bdist_wheel for cryptography: finished with status 'error'
    Complete output from command /home/user/PyVE/Repo/Leo/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5wtaw6vz/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-u1dxni1j --python-tag cp37:
    running bdist_wheel
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/cryptography
    copying src/cryptography/__about__.py -> build/lib.linux-x86_64-3.7/cryptography
    copying src/cryptography/exceptions.py -> build/lib.linux-x86_64-3.7/cryptography
    copying src/cryptography/fernet.py -> build/lib.linux-x86_64-3.7/cryptography
    copying src/cryptography/__init__.py -> build/lib.linux-x86_64-3.7/cryptography
    copying src/cryptography/utils.py -> build/lib.linux-x86_64-3.7/cryptography
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat
    copying src/cryptography/hazmat/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
    copying src/cryptography/hazmat/_der.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
    copying src/cryptography/hazmat/_oid.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
    copying src/cryptography/hazmat/_types.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
    creating build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/ocsp.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/name.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/oid.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/general_name.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/base.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/extensions.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    copying src/cryptography/x509/certificate_transparency.py -> build/lib.linux-x86_64-3.7/cryptography/x509
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings
    copying src/cryptography/hazmat/bindings/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/_asymmetric.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/_cipheralgorithm.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/keywrap.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/hmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/hashes.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/poly1305.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/padding.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/_serialization.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/constant_time.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    copying src/cryptography/hazmat/primitives/cmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/backends
    copying src/cryptography/hazmat/backends/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends
    copying src/cryptography/hazmat/backends/interfaces.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
    copying src/cryptography/hazmat/bindings/openssl/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
    copying src/cryptography/hazmat/bindings/openssl/_conditional.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
    copying src/cryptography/hazmat/bindings/openssl/binding.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
    copying src/cryptography/hazmat/primitives/ciphers/aead.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
    copying src/cryptography/hazmat/primitives/ciphers/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
    copying src/cryptography/hazmat/primitives/ciphers/base.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
    copying src/cryptography/hazmat/primitives/ciphers/modes.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
    copying src/cryptography/hazmat/primitives/ciphers/algorithms.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/ed448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/ec.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/dsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/x448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/x25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/rsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/ed25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/dh.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/padding.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    copying src/cryptography/hazmat/primitives/asymmetric/utils.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/kbkdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/scrypt.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/hkdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/pbkdf2.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/concatkdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    copying src/cryptography/hazmat/primitives/kdf/x963kdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
    copying src/cryptography/hazmat/primitives/serialization/pkcs12.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
    copying src/cryptography/hazmat/primitives/serialization/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
    copying src/cryptography/hazmat/primitives/serialization/base.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
    copying src/cryptography/hazmat/primitives/serialization/pkcs7.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
    copying src/cryptography/hazmat/primitives/serialization/ssh.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
    copying src/cryptography/hazmat/primitives/twofactor/totp.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
    copying src/cryptography/hazmat/primitives/twofactor/hotp.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
    copying src/cryptography/hazmat/primitives/twofactor/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
    copying src/cryptography/hazmat/primitives/twofactor/utils.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
    creating build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/ocsp.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/x509.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/ed448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/aead.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/ec.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/dsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/backend.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/x448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/decode_asn1.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/x25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/ciphers.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/hmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/rsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/hashes.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/ed25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/poly1305.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/dh.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/utils.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/encode_asn1.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    copying src/cryptography/hazmat/backends/openssl/cmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
    running egg_info
    writing src/cryptography.egg-info/PKG-INFO
    writing dependency_links to src/cryptography.egg-info/dependency_links.txt
    writing requirements to src/cryptography.egg-info/requires.txt
    writing top-level names to src/cryptography.egg-info/top_level.txt
    reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    no previously-included directories found matching 'docs/_build'
    warning: no previously-included files found matching 'vectors'
    warning: no previously-included files matching '*' found under directory 'vectors'
    warning: no previously-included files matching '*' found under directory '.github'
    warning: no previously-included files found matching 'release.py'
    warning: no previously-included files found matching '.coveragerc'
    warning: no previously-included files found matching 'codecov.yml'
    warning: no previously-included files found matching '.readthedocs.yml'
    warning: no previously-included files found matching 'dev-requirements.txt'
    warning: no previously-included files found matching 'tox.ini'
    warning: no previously-included files found matching 'mypy.ini'
    warning: no previously-included files matching '*' found under directory '.zuul.d'
    warning: no previously-included files matching '*' found under directory '.zuul.playbooks'
    writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
    copying src/cryptography/py.typed -> build/lib.linux-x86_64-3.7/cryptography
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.7/_padding.c'
    creating build/temp.linux-x86_64-3.7
    generating cffi module 'build/temp.linux-x86_64-3.7/_openssl.c'
    running build_rust
  
        =============================DEBUG ASSISTANCE=============================
        If you are seeing a compilation error please try the following steps to
        successfully install cryptography:
        1) Upgrade to the latest pip and try again. This will fix errors for most
           users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
        2) Read https://cryptography.io/en/latest/installation.html for specific
           instructions for your platform.
        3) Check our frequently asked questions for more information:
           https://cryptography.io/en/latest/faq.html
        4) Ensure you have a recent Rust toolchain installed:
           https://cryptography.io/en/latest/installation.html#rust
        5) If you are experiencing issues with Rust for *this release only* you may
           set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
        =============================DEBUG ASSISTANCE=============================
  
    error: can't find Rust compiler
  
    If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
  
    To update pip, run:
  
        pip install --upgrade pip
  
    and then retry package installation.
  
    If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
  
    This package requires Rust >=1.41.0.
  
    ----------------------------------------
    Failed building wheel for cryptography
    Running setup.py clean for cryptography
    Complete output from command /home/user/PyVE/Repo/Leo/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5wtaw6vz/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" clean --all:
    running clean
    removing 'build/temp.linux-x86_64-3.7' (and everything under it)
    removing 'build/lib.linux-x86_64-3.7' (and everything under it)
    'build/bdist.linux-x86_64' does not exist -- can't clean it
    'build/scripts-3.7' does not exist -- can't clean it
    removing 'build'
    running clean_rust
  
        =============================DEBUG ASSISTANCE=============================
        If you are seeing a compilation error please try the following steps to
        successfully install cryptography:
        1) Upgrade to the latest pip and try again. This will fix errors for most
           users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
        2) Read https://cryptography.io/en/latest/installation.html for specific
           instructions for your platform.
        3) Check our frequently asked questions for more information:
           https://cryptography.io/en/latest/faq.html
        4) Ensure you have a recent Rust toolchain installed:
           https://cryptography.io/en/latest/installation.html#rust
        5) If you are experiencing issues with Rust for *this release only* you may
           set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
        =============================DEBUG ASSISTANCE=============================
  
    error: can't find Rust compiler
  
    If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
  
    To update pip, run:
  
        pip install --upgrade pip
  
    and then retry package installation.
  
    If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
  
    This package requires Rust >=1.41.0.
  
    ----------------------------------------
    Failed cleaning build dir for cryptography
  Failed to build cryptography
  Installing collected packages: setuptools, setupext-janitor, wheel, certifi, chardet, urllib3, idna, requests, requests-toolbelt, pkginfo, rfc3986, typing-extensions, zipp, importlib-metadata, jeepney, pycparser, cffi, cryptography, SecretStorage, keyring, tqdm, docutils, pyparsing, packaging, six, webencodings, bleach, Pygments, readme-renderer, colorama, twine
    Running setup.py install for cryptography: started
      Running setup.py install for cryptography: finished with status 'error'
      Complete output from command /home/user/PyVE/Repo/Leo/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5wtaw6vz/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-wkvenu2e/install-record.txt --single-version-externally-managed --prefix /tmp/pip-build-env-_mtbzqd1 --compile --install-headers /home/user/PyVE/Repo/Leo/include/site/python3.7/cryptography:
      running install
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-3.7
      creating build/lib.linux-x86_64-3.7/cryptography
      copying src/cryptography/__about__.py -> build/lib.linux-x86_64-3.7/cryptography
      copying src/cryptography/exceptions.py -> build/lib.linux-x86_64-3.7/cryptography
      copying src/cryptography/fernet.py -> build/lib.linux-x86_64-3.7/cryptography
      copying src/cryptography/__init__.py -> build/lib.linux-x86_64-3.7/cryptography
      copying src/cryptography/utils.py -> build/lib.linux-x86_64-3.7/cryptography
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat
      copying src/cryptography/hazmat/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
      copying src/cryptography/hazmat/_der.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
      copying src/cryptography/hazmat/_oid.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
      copying src/cryptography/hazmat/_types.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat
      creating build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/ocsp.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/name.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/oid.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/general_name.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/base.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/extensions.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      copying src/cryptography/x509/certificate_transparency.py -> build/lib.linux-x86_64-3.7/cryptography/x509
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings
      copying src/cryptography/hazmat/bindings/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/_asymmetric.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/_cipheralgorithm.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/keywrap.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/hmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/hashes.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/poly1305.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/padding.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/_serialization.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/constant_time.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      copying src/cryptography/hazmat/primitives/cmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/backends
      copying src/cryptography/hazmat/backends/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends
      copying src/cryptography/hazmat/backends/interfaces.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
      copying src/cryptography/hazmat/bindings/openssl/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
      copying src/cryptography/hazmat/bindings/openssl/_conditional.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
      copying src/cryptography/hazmat/bindings/openssl/binding.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/bindings/openssl
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
      copying src/cryptography/hazmat/primitives/ciphers/aead.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
      copying src/cryptography/hazmat/primitives/ciphers/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
      copying src/cryptography/hazmat/primitives/ciphers/base.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
      copying src/cryptography/hazmat/primitives/ciphers/modes.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
      copying src/cryptography/hazmat/primitives/ciphers/algorithms.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/ciphers
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/ed448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/ec.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/dsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/x448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/x25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/rsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/ed25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/dh.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/padding.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      copying src/cryptography/hazmat/primitives/asymmetric/utils.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/asymmetric
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/kbkdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/scrypt.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/hkdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/pbkdf2.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/concatkdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      copying src/cryptography/hazmat/primitives/kdf/x963kdf.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/kdf
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
      copying src/cryptography/hazmat/primitives/serialization/pkcs12.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
      copying src/cryptography/hazmat/primitives/serialization/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
      copying src/cryptography/hazmat/primitives/serialization/base.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
      copying src/cryptography/hazmat/primitives/serialization/pkcs7.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
      copying src/cryptography/hazmat/primitives/serialization/ssh.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/serialization
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
      copying src/cryptography/hazmat/primitives/twofactor/totp.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
      copying src/cryptography/hazmat/primitives/twofactor/hotp.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
      copying src/cryptography/hazmat/primitives/twofactor/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
      copying src/cryptography/hazmat/primitives/twofactor/utils.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/primitives/twofactor
      creating build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/ocsp.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/x509.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/ed448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/aead.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/ec.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/dsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/backend.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/x448.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/decode_asn1.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/x25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/ciphers.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/hmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/rsa.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/__init__.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/hashes.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/ed25519.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/poly1305.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/dh.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/utils.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/encode_asn1.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      copying src/cryptography/hazmat/backends/openssl/cmac.py -> build/lib.linux-x86_64-3.7/cryptography/hazmat/backends/openssl
      running egg_info
      writing src/cryptography.egg-info/PKG-INFO
      writing dependency_links to src/cryptography.egg-info/dependency_links.txt
      writing requirements to src/cryptography.egg-info/requires.txt
      writing top-level names to src/cryptography.egg-info/top_level.txt
      reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      no previously-included directories found matching 'docs/_build'
      warning: no previously-included files found matching 'vectors'
      warning: no previously-included files matching '*' found under directory 'vectors'
      warning: no previously-included files matching '*' found under directory '.github'
      warning: no previously-included files found matching 'release.py'
      warning: no previously-included files found matching '.coveragerc'
      warning: no previously-included files found matching 'codecov.yml'
      warning: no previously-included files found matching '.readthedocs.yml'
      warning: no previously-included files found matching 'dev-requirements.txt'
      warning: no previously-included files found matching 'tox.ini'
      warning: no previously-included files found matching 'mypy.ini'
      warning: no previously-included files matching '*' found under directory '.zuul.d'
      warning: no previously-included files matching '*' found under directory '.zuul.playbooks'
      writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
      copying src/cryptography/py.typed -> build/lib.linux-x86_64-3.7/cryptography
      running build_ext
      generating cffi module 'build/temp.linux-x86_64-3.7/_padding.c'
      creating build/temp.linux-x86_64-3.7
      generating cffi module 'build/temp.linux-x86_64-3.7/_openssl.c'
      running build_rust
  
          =============================DEBUG ASSISTANCE=============================
          If you are seeing a compilation error please try the following steps to
          successfully install cryptography:
          1) Upgrade to the latest pip and try again. This will fix errors for most
             users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
          2) Read https://cryptography.io/en/latest/installation.html for specific
             instructions for your platform.
          3) Check our frequently asked questions for more information:
             https://cryptography.io/en/latest/faq.html
          4) Ensure you have a recent Rust toolchain installed:
             https://cryptography.io/en/latest/installation.html#rust
          5) If you are experiencing issues with Rust for *this release only* you may
             set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
          =============================DEBUG ASSISTANCE=============================
  
      error: can't find Rust compiler
  
      If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
  
      To update pip, run:
  
          pip install --upgrade pip
  
      and then retry package installation.
  
      If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
  
      This package requires Rust >=1.41.0.
  
      ----------------------------------------
  Command "/home/user/PyVE/Repo/Leo/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-5wtaw6vz/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-wkvenu2e/install-record.txt --single-version-externally-managed --prefix /tmp/pip-build-env-_mtbzqd1 --compile --install-headers /home/user/PyVE/Repo/Leo/include/site/python3.7/cryptography" failed with error code 1 in /tmp/pip-install-5wtaw6vz/cryptography/
  
  ----------------------------------------
Command "/home/user/PyVE/Repo/Leo/bin/python3 -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-_mtbzqd1 --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools>=38.6.0 setupext-janitor>=1.1 wheel>=0.31.0 twine>=1.11.0" failed with error code 1 in None
`(Leo) user@debian-leo-vm:~/PyVE/Repo/Leo/leo-editor$ `

**Comment by ranvik14** [comment]
@edreamleo  & @maphew : Please let me know, if there's anything else that I should provide.

**Comment by ranvik14** [comment]
@edreamleo : My intention was only to report this issue - but - I did not consider myself being the right one resolving this issue.

Anyhow, as long as @maphew does not re-assign this issue to himself, I'll try to spend some time on finding a resolution for it.

**Comment by tbpassin** [comment]
Looks like a Rust compiler needed to be installed (I imagine you noticed that already :) ).

**Comment by tbpassin** [comment]
I'm going to make much the same suggestion here that I did on the Google Groups thread.  If you want to run a developmental version of Leo - and you wrote that this situation would be for developers - then  you don't really need to "install" it at all.  You can run it from a clone of the Github repo.  You check out the branch you want, and then use the PYTHONPATH environmental variable to point to the checked-out branch of your clone.

If you need a new dependency, you will get an error message about it and you can install it using pip.

You can make a desktop launcher by writing a shell script to set the PYTHONPATH and launch Leo.  The desktop launcher would then launch the shell script.  Or just use the shell script itself without a desktop launcher.

I tried this on Debian and didn't have any problems getting it working.  I used git-cola as a git GUI client, instead of just using raw git commands in a terminal, BTW.  For plain checkouts, this was probably overkill.

**Comment by ranvik14** [comment]
@tbpassin : Yes, I noticed the need for a Rust compiler - and - thanks for your other follow-up.

I'll re-read the conversation on Google Groups - as well as - all the details available for #1847 and #1848 (this one).

**Comment by ranvik14** [comment]
@tbpassin : I've done, what I stated yesterday - but - I'm still confused about the 'real' problem :-(

Would you be able to provide a fix for #1847 and / or #1848 (this one) from your side for Debian?

**Comment by tbpassin** [comment]
https://github.com/leo-editor/leo-editor/issues/1847 would seem to be a case of some dependency missing from the requirements list of Python 6.3.  Installing 6.1 first  presumably provides the missing bit.  I know nothing about the way in which the requirement list/installation details are put together, so I'm not the best person for this.

I'm also not enthusiastic about working on https://github.com/leo-editor/leo-editor/issues/1848.  I don't see a need to make a pip-installation.  Even if you did pip-install from say the devel branch, it's changing so fast that you'd be out of date in days or weeks.  So you'd want to re-install anyway.  I think that pointing PYTHONPATH to a git clone is a better way to go.

Of course, *someone* at some point needs to make sure a new release can install properly, but that doesn't seem to be what you are asking for.

If you still don't agree with what I've said, would you please say more so that we can understand better why you want something else?

**Comment by ranvik14** [comment]
@tbpassin : I fully understand your last comment.

That is I don't feel as the right persons to take over responsibility for delivering / providing a version of 'leo-stable' for Debian on PyPI - and -  I don't know either how a version of 'leo-devel' is created on Debian from Git/ GitHub.

The reason, why I raised both issues as a problem for Leo in the first place, is that 'PKG-INFO.TXT' does state that 'Linux' is a supported platform beside 'MacOS' & 'Windows' ...

I don't know if - or - how it is possible to further clarify / select the supported Linux Distributions like Debian, Fedora & OpenSUSE, etc. which Leo is checking actively before releasing a new version to the 'leo-stable' or ' leo-devel' branch.

The only thing I do know is that Leo is working w/o any issues known to me in a Fedora Linux environment ;-)

**Comment by tbpassin** [comment]
So if we stay with your breakdown of uses, there is some probably minor packaging problem of the 6.3 release that caused some problem on Debian, but when worked around with the "install 6.1 first" hack, Leo runs as expected on Debian.  Undesirable, but minor.

Then there is the question of running a developmental version of Leo on some version or other of Linux.  You wrote "before releasing a new version to the ... 'leo-devel' branch.  But that branch is specifically for development by Leo's developers, and can change all the time.   The devel branch is not considered any kind of "release".  It's just a snapshot of whatever developmental changes have been merged into the branch at that moment.

I usually run from the devel branch even if I'm not developing something for Leo, because it has more recent versions of certain software, or certain bug fixes that I want to be in there.  But sometimes there are problems with the devel branch, and so if need be I go back to running the 6.3 release version.

Even with continuous integration, which Leo uses when merging pull requests, something in development could break at any time.  That doesn't usually happen, but it can.

So I am not concerned that some packaging problem might prevent the devel branch from "installing" on some random version of Linux.  Of course, installation has to be gotten to work for an actual release.

Anyway, as I recall the problem you had on Debian with the devel branch was that your computer didn't have Rust on it.  That seemed to be needed for some crypto package to build.  Leo isn't going to be installing the Rust system for you.

You wrote " I don't know either how a version of 'leo-devel' is created on Debian from Git/ GitHub".  I don't know, either, but I am sure that there has never been a "Debian-specific" version released by Leo.  It's possible, and may have happened, that someone sometime created a Debian install package.  They might rearrange the locations of some libraries to match how Debian Python installations are set up, for example.  But Leo hasn't been making any such packages, so far as I know.

**Comment by tbpassin** [comment]
All right, I have got this worked out.  The problem with installing Leo from a git clone was that the installer wanted to invoke a Rust compiler to build the cryptography module.  Turns out, if Leo was previously installed, this module was already installed.  

The instructions at [Installing Leo](http://leoeditor.com/installing.html) worked, but it seems that pip has to be run using _sudo_ (because some of the packages were previously installed into the all-user location instead of the local user, I think):
~~~
cd git
sudo python3 -m pip install --editable leo-editor
~~~
It may possibly be necessary to upgrade pip, and if so I had to do it with sudo again.

Now, I did get a warning installing with _sudo_:

~~~
WARNING: Running pip as root will break packages and permissions. 
                You should install packages reliably by using 
                venv: https://pip.pypa.io/warnings/venv
~~~
But, ignoring the warning, after this Leo did launch using the _devel_ branch (or whatever branch has been checked out).  

So I did succeed in installing Leo on Debian to run from the git clone.  BTW, I also did install Rust and tried to install/build the cryptography module.  Once again, I had to run with _sudo_.  But then I noticed that the cryptography module had already been installed, and so it wasn't necessary to be able to build it.  So the Rust install wasn't needed after all.  I verified this by reverting my whole Debian installation (I had saved a snapshot before installing Rust) and going through the steps above.

On my Debian system, the installed python3.7 modules are in
~~~
/usr/local/lib/python3.7/dist-packages
~~~
(that is, not in _site-packages_ but _dist-packages_).  I checked the file permissions on some of the modules and they were readable by anyone.  So presumably if you had some permissions problem after _sudo pip_-installing something (like Leo), you could reset the directory so all files are readable by everyone once again.

**Comment by tbpassin** [comment]
I wrote "...presumably if you had some permissions problem after sudo pip-installing something (like Leo), you could reset the directory so all files are readable by everyone once again."  That would be _read_ and _write_ both, ortherwise pip wouldn't be able to update anything.

**Comment by ranvik14** [comment]
@tbpassin : ### Thanks a lot for the detailed explanations & findings you documented here.

I will not have time before the weekend to spend the necessary time to execute them in a new Debian virtual environment of my own - but - I promise I'll report back here ...

FYI: I do understand, that the purpose of 'leo-devel' is to just provide the current / latest version of what Edward eventually will turn into 'leo-stable' - and - that no version from it will be uploaded to PyPI at all.

Where I do have a somewhat different perception of what you have stated, is that 'leo-devel' is recommended to be used only by Leo's developers. - I've seen in multiple locations & multiple times, that the usage of 'leo-devel' is the recommended way for regular users to get bug fixed before a new stable version is released.

And this is the area / aspect where I believe your & my expectations are different. - If the 'leo-devel' branch is recommended also for regular users I would expect / hope, that Leo is tested on a rpm- (e.g. Fedora / OpenSUSE) - as well as - on a deb-based (e.g. Debian / Ubuntu) system before merging a PR into it.

FWIW: I might be completely wrong - but - I hope that @edreamleo is commenting on these different interpretations & views as well.

**Comment by tbpassin** [comment]
Well, yes, bugs may get fixed in some versions of the _devel_ branch.   The next day there may be code that needs some experience in the real world since it might not work right in all cases.  I don't know for sure what the continuous integration server runs, but I am pretty sure it's not installations.  Mainly it runs the Leo test site and makes sure nothing is broken by the latest merge operation.  Edward or someone else who knows the process better can speak more to this.

If there is a working Leo installation, then generally a later version of the devel branch will work too, and on any platform.  The only exception would be if a higher version of some library, like the Qt libraries, were needed.  That wouldn't happen for someone pulling from the latest _devel_ branch.

The _devel_ branch also may contain new functionality, and someone may want that.  The Viewrendered3 plugin has been advancing that way.

So yes, there can be advantages to using the _devel_ branch of Leo.  But it may not always be stable.  And it is unlikely to have been tested to see if a from-scratch installation on some random platform will work. I would say that if someone wants to start using Leo, they should install some release version, like 6.3.  That will get the dependencies installed, and let the user make sure everything works.  Then they can try the _devel_ version if they want, in one of the ways that have been discussed.

**Comment by tbpassin** [comment]
There **are** some Debian-specific installation issues, not with Leo specifically, because Debian splits up the parts of a Python installation in a non-standard way, and pip can have some trouble with it.  See

[Pip Issue 9617](https://github.com/pypa/pip/issues/9617)

This issue is not unique to Debian, and apparently needs to be worked out between the Pip folks and the Debian folks.  This may be mostly a problem for future versions of Pip, when they transition away from the current way of locating packages (_distutils_, which is being deprecated) and move to a new one.

**Comment by ranvik14** [comment]
@tbpassin : I'm not able to reproduce your findings - but - at least I understand why ;-)

I'm doing all my work in VMs on top of the Qubes OS [a], which uses so called Template VMs [b], which are then used to create the user-facing instances of Application VMs.

All content from the used Template VM is available read-only in the using Application VMs ...

When I follow your example steps Leo from GitHub installs - but - again fails to start with:

```
    user@debian-leo-vm:~$ 
    user@debian-leo-vm:~$ pip3 show leo
    Name: leo
    Version: 6.3.dev1082
    Summary: An IDE, PIM and Outliner
    Home-page: http://leoeditor.com
    Author: Edward K. Ream
    Author-email: edreamleo@gmail.com
    License: MIT License
    Location: /home/user/Test-Repo2-for-Leo/leo-editor
    Requires: PyQt5, PyQtWebEngine, asttokens, docutils, flexx, meta, nbformat, pylint, pyflakes, black, pyshortcuts, sphinx
    Required-by: 
    user@debian-leo-vm:~$ leo&
    [1] 4489
    user@debian-leo-vm:~$ 
    setting leoID from os.getenv('USER'): 'user'
    qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
    This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
    
    Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.
    
    
    [1]+  Aborted                 leo
    user@debian-leo-vm:~$ 

```

I'll investigate further - but - I do understand that my setup is now a special case of Pip Issue 9617.

---

[a] Qubes OS - A reasonably secure operating system

* https://www.qubes-os.org/intro/

[b] TemplateVMs

* https://www.qubes-os.org/doc/templates/

**Comment by tbpassin** [comment]
I have hit that problem once or twice - the xcb issue.  I don't remember the fix, but I did find one on line.  So a search for that error message should turn up something useful.

**Comment by tbpassin** [comment]
This link may do the job for you -

[How to fix 'Could not load the Qt platform plugin “xcb” in “” even though it was found.' after fixup_bundle macro?
](https://stackoverflow.com/questions/57362015/how-to-fix-could-not-load-the-qt-platform-plugin-xcb-in-even-though-it-was)

**Comment by ranvik14** [comment]
@tbpassin : Thanks a lot for this info & link. - At the moment it's not (yet) helpful to me, since its Qt platform specific, which is something I've never looked into so far ...

I decided to concentrate more on what has changed b/w Leo 6.1 and later, since as documented in #1847,  version 6.1 is the last version of Leo, which installs w/o an issue on Debian ...

What I found out so far, is that with version 6.2 Leo has switched from specifying the dependencies explicitly in 'requirements.txt' to setting it to "--editable ." and changing the 'universal' flag in 'setup.cfg' from 1 to 0.

I guess I now should start reading the details of 'Packaging and distributing projects' from the Python Packaging User Guide. - See

* https://packaging.python.org/guides/distributing-packages-using-setuptools/#working-in-development-mode

**Comment by tbpassin** [comment]
Yes, it is Qt-specific, but that is a (or *the*) key error in the output you included.  Also, I hit this problem once myself (I forget which Linux distro it was).  And it's a problem that (I think) Leo can't really do anything about on its own.

Leo asks for Qt (really PyQt) to get installed.  If that installation process doesn't work, it's a PyQt problem, not Leo's.  But I never had any problem show up when installing Leo on Linux except for that "install 6.1 first" thing).  I agree you can probably make progress on that, but it won't solve the missing .so shared library so far as I can see - which might not be very far.

**Comment by ranvik14** [comment]
@tbpassin : I finally did make some progress. - I changed the version range for the packages 'PyQt5' as well as 'PyQtWebEngine' in "setup.py" to use the same / similar values like for Leo 6.1.

```
user_requires = [
    'PyQt5 >= 5.12, < 5.13',  # VR: Use similar versions like for Leo 6.1. # v5.12+ to close #1217
    'PyQtWebEngine >= 5.12, < 5.13',  # VR: Use similar versions like for Leo 6.1. #1202 QtWebKit needs to be installed separately starting Qt 5.6
    'asttokens',  # abstract syntax tree text parsing
    'docutils',  # used by Sphinx, rST plugin
    'flexx',  # for LeoWapp browser gui
    'meta',  # for livecode.py plugin, which is enabled by default
    'nbformat',  # for Jupyter notebook integration
    'pylint', 'pyflakes', 'black',  # coding syntax standards
    'pyshortcuts >= 1.7',  # desktop integration (#1243)
    'sphinx',  # rST plugin
    'windows-curses; platform_system=="Windows"',  # for console mode on Windows
    ]

```

I still don't know what's the root cause - but - some progress ;-)

**Comment by tbpassin** [comment]
Nice work getting on top of that setup stuff.

I don't like the low PyQt5 version, though.  I've already had cases where I had PyQt5 v5.15 installed for one package, and when I installed another it got downgraded to 5.13.  I'd rather that Leo not be one of those offenders.

**Comment by ranvik14** [comment]
@tbpassin : I did investigate a bit further - and - I guess it will be only with Debian 11 (aka Bullseye) that PyQt 5.15 will be officially supported by the Debian community.

I'm kind of tempted to propose to close both this issue as well as #1847 with an explanation of the workaround - but - also a clear statement, that it will not be fixed. - Neither in the 'master' nor in the 'devel' branch. - WDYT?

**Comment by tbpassin** [comment]
QT5 versions beyond 5.15 will apparently only be available to paying QT customers - this to supposedly because new development effort will be going only into QT 6+.  So we probably won't get PyQt5 beyond v5.15.

Given that, and the fact that Leo works with 5.15 (not counting Debian - grrr), I would go with your suggestion.  I wonder whether the requirements list can be used to change the PyQt version when it's installing on Debian.  We know that there can be a windows conditional, so maybe this would be possible.

Otherwise, where would the warning go so that a user, installing for the first time, could see it and make sense of it?

**Comment by ranvik14** [comment]
@tbpassin : Thanks for your feedback! - I've asked @edreamleo for his view on #1847 . - Depending on his answer we can decide what still should be done for this issue in the context of Leo 6.4 / the devel branch.

**Comment by edreamleo** [comment]
I am going to mark this as "Won't Fix".

**Comment by ranvik14** [comment]
@edreamleo : After your recent changes to 'setup.py' etc., I can report that this issue is fixed now as well. - Leo's 'devel' branch can successfully be installed on Debian & is working w/o any issues.

**Comment by edreamleo** [comment]
@ranvik14 Thanks for this confirmation.
